arith/mersenne31/src/m31x16/m31_avx256.rs

```diff
@@ -284,16 +284,48 @@ impl SimdField for AVXM31 {
 
     const PACK_SIZE: usize = M31_PACK_SIZE;
 
+    #[inline(always)]
     fn pack(base_vec: &[Self::Scalar]) -> Self {
         assert_eq!(base_vec.len(), M31_PACK_SIZE);
         let ret: [Self::Scalar; M31_PACK_SIZE] = base_vec.try_into().unwrap();
         unsafe { transmute(ret) }
     }
 
+    #[inline(always)]
     fn unpack(&self) -> Vec<Self::Scalar> {
         let ret = unsafe { transmute::<[__m256i; 2], [Self::Scalar; M31_PACK_SIZE]>(self.v) };
         ret.to_vec()
     }
+
+    #[inline(always)]
+    fn horizontal_sum(&self) -> Self::Scalar {
+        let mut buffer = 0u64;
+        let ret = unsafe { transmute::<[__m256i; 2], [Self::Scalar; M31_PACK_SIZE]>(self.v) };
+
```

Comment:
let buffer = ret.iter().map(|r|r.v as u64).sum();

Reply:
sure, see https://github.com/PolyhedraZK/Expander/pull/234/commits/084fca8fe01317ba4287fc5bf47614a01850db5d

I was trying to do loop unrolling manually, but it seems as good I guess?